### PR TITLE
 Convert github to use non-factory resource

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Per internal discussion we are moving aware from ConfigurableResourceFactory for our integrations. This moves Github.

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13480** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13480" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13480?utm_source=stack-comment).

File: python_modules/libraries/dagster-github/dagster_github/resources.py [hunk 1]
```diff
@@ -185,7 +180,7 @@ class GithubResource(ConfigurableResourceFactory[GithubClient]):
         ),
     )
 
-    def create_resource(self, context: InitResourceContextWithKeyMapping) -> GithubClient:
+    def get_github_client(self) -> GithubClient:
```
Comment (benpankow) [inline python_modules/libraries/dagster-github/dagster_github/resources.py:183]: `get_client`? what pattern do we want to follow here?

**Review by benpankow — APPROVED** [review]
lgtm, see question about `get_github_client`

File: python_modules/libraries/dagster-github/dagster_github/resources.py (hunk 1 above)
Comment (schrockn) [reply python_modules/libraries/dagster-github/dagster_github/resources.py:183]: IMO when there is a ClassName we should have a method `get_class_name()`

Comment (schrockn) [reply python_modules/libraries/dagster-github/dagster_github/resources.py:183]: However I wasn't consistent there as it should have been `get_datadog_client` in the datadog diff

**Pull request merged**